ios/browser/api/bookmarks/brave_bookmarks_api.mm

```diff
@@ -51,6 +52,26 @@ - (instancetype)initWithNode:(IOSBookmarkNode*)node
   }
   return self;
 }
+
+- (BOOL)isEqual:(id)object {
+  if (self == object) {
+    return YES;
+  }
+
+  if (![object isKindOfClass:[BookmarkFolder class]]) {
+    return NO;
+  }
+
+  BookmarkFolder* other = (BookmarkFolder*)object;
+
+  return self.bookmarkNode.nodeId == other.bookmarkNode.nodeId &&
+         self.indentationLevel == other.indentationLevel;
+}
+
+- (NSUInteger)hash {
+  return base::HashCombine(0ull, [self.bookmarkNode hash],
```

Comment:
Should match equality check, meaning this should be hashing `nodeId` not the entire node, or equality check should be doing an equality check against the entire node